### DISCUSSION TITLE
Editing a private repository without giving the users access

### DISCUSSION THREAD

**Discussion by rmortes** [sveltia/sveltia-cms#261] [Q&A]
Is there a way to allow a third party to make changes to a website using this CMS without giving them direct access to the repository? Like, using a proxy to make the changes for them or something like that?

The repo is in gitlab, but I could move it to github if needed. Also, the repo being private is required.

Thanks!

**Comment by kyoshino** [comment]
In [Netlify/Decap CMS](https://decapcms.org/docs/choosing-a-backend/), the Git Gateway backend works as a proxy in combination with the Netlify Identity widget. Unfortunately it’s very slow, so Sveltia CMS will not support the combo as mentioned in the [compatibility info](https://github.com/sveltia/sveltia-cms?tab=readme-ov-file#compatibility), but I have a plan to implement similar proxy functionality in the future. No ETA yet 😉 

> The Bitbucket, Gitea/Forgejo and Git Gateway backends will not be supported for performance reasons. We may implement a high-performance Git Gateway alternative in the future.

**Reply by rmortes** [reply]
Great to know! Sorry about opening an issue, I didn't notice you were using discussions. I'd like to give it a go and try to work on an implementation of a backend for this purpose, how do I go about this? Is there a way to extend the sveltia backend catalogue with a plugin, or should I fork the project to work on a new backend?

Thanks!

**Reply by kyoshino** [reply]
No worries. There is no plugin system yet, so you need to fork Sveltia CMS if you want to implement a proxy sooner. Note that you need a server side implementation as well.

Otherwise, you can stick with Decap CMS + Git Gateway + Netlify Identity. It’s slow but should work with GitLab.

**Reply by rmortes** [reply]
If at all possible I'd like to not use a proprietary solution like Netlify Identity.

Regarding the server side API, yeah, I figured as much by looking at the current backend implementation. I'll let you know if I come up with something 👍

**Reply by kyoshino** [reply]
Just FYI: the core of Netlify Identity is open source: [GoTrue](https://github.com/netlify/gotrue)

There is a fork of it: [Supabase Auth](https://github.com/supabase/auth)

[Git Gateway](https://github.com/netlify/git-gateway) itself is also open source.
